x/tss/keeper/genesis.go

```diff
@@ -84,6 +84,30 @@ func (k Keeper) InitGenesis(ctx sdk.Context, snapshotter types.Snapshotter, genS
 
 		k.setExternalKeys(ctx, externalKeys)
 	}
+
+	for _, signature := range genState.Signatures {
+		if signature.SigStatus != exported.SigStatus_Signed {
+			panic(fmt.Errorf("signature %s is not completed", signature.SigID))
+		}
+
+		if signature.GetSingleSig() == nil && signature.GetMultiSig() == nil {
+			panic(fmt.Errorf("signature %s is not completed", signature.SigID))
+		}
+
+		if _, ok := k.getSig(ctx, signature.SigID); ok {
+			panic(fmt.Errorf("signature %s already set", signature.SigID))
+		}
+
+		k.SetSig(ctx, signature)
+	}
+
+	for _, validatorStatus := range genState.ValidatorStatuses {
+		if _, ok := k.getValidatorStatus(ctx, validatorStatus.Validator); ok {
+			panic(fmt.Errorf("validator status %s already set", validatorStatus.Validator.String()))
+		}
```

Comment:
I kinda decided that letting keeper check for uniqueness is easier and also thought it's more efficient, unless you disagree?

Reply:
I don't think we need to worry that much about performance for import and export, since it is rarely done and not time critical. the benefit of doing it in Validate is that you can return an error instead of panicking. I don't have a good intuition yet for which way is better, so i don't fell strongly about it

Reply:
Let's discuss and unify later.